tidy3d/web/api/container.py

```diff
@@ -209,13 +209,19 @@ def task_id(self) -> TaskId:
         """The task ID for this ``Job``. Uploads the ``Job`` if it hasn't already been uploaded."""
         if self.task_id_cached:
             return self.task_id_cached
-        return self.upload()
+        return self._upload()
 
-    def upload(self) -> TaskId:
+    def _upload(self) -> TaskId:
+        """Upload this job and return the task ID for handling."""
         # upload kwargs with all fields except task_id
         self_dict = self.dict()
         upload_kwargs = {key: self_dict.get(key) for key in self._upload_fields}
-        return web.upload(**upload_kwargs)
+        task_id = web.upload(**upload_kwargs)
+        return task_id
+
+    def upload(self) -> None:
```

Comment:
It should probably still return the task_id?

Reply:
I dont think it ever does for `Job.upload()`.

Reply:
the user can always grab `job.task_id`, but the task id isn't returned. at least that's how it was before and I think generally makes sense?

Reply:
oh I guess there never was a `Job.upload` before 2.7.. and the old version I had implemented did return task_id. But I think in general it should not. since it's already in the job?

Reply:
No, previously it would return it, and I think that makes sense since `web.upload` returns it.

https://github.com/flexcompute/tidy3d/blob/4918797e504db4677c064f682a9335c48a538471/tidy3d/web/api/container.py#L218

Reply:
yea I think this old version was not working quite properly though. so this was a bit of a hack. The new approach seems reasonable to me and we can choose to return it or not. If you want to return task_id, it shouldn't affect  anything and I don't care strongly one way or the other, feel free to do whatever before squashing and merging.